app/models/bag_man_request.rb

```diff
@@ -39,7 +39,7 @@ def cancel!(reason)
     unless cancelled
       transaction do
         update!(cancelled: true, cancel_reason: reason)
-        replication_transfer.cancel!(reason)
+        replication_transfer.cancel!(reason, last_error)
         send_cancel(replication_transfer)
       end
     end
```

Comment:
OK, sounds like it should be in the transaction.  Although, just adding an item to the job queue doesn't ensure that it's done.  So, wondering whether the job itself should be using the API to check or update the cancel data?  But then, if the job does issue a cancel API request, this could incur another level of loop management?